src/Jank/TermManagement/reducer.ts

```diff
@@ -0,0 +1,23 @@
+import { Term, Action, ActionType } from './types';
+
+/**
+ * State reducer for terms, handling an action that was dispatched.
+ *
+ * @param state the current state, containing the list of terms
+ * @param action the dispatched action to handle
+ * @returns the new state after the dispatched action was handled
+ */
+const reducer = (state: Term[], action: Action) => {
```

Comment:
This is the reducer that keeps the current state (the first parameter) and the current action (second parameter). The reducer is called whenever an action is dispatched, and returns the updated state after handling the action. The updated state causes a re-render in the component, so the UI will be updated without any explicit trigger - React handles that for us.